pkg/limiter/fixed_test.go

```diff
@@ -1,26 +1,46 @@
 package limiter_test
 
 import (
+	"context"
 	"testing"
+	"time"
 
 	"github.com/influxdata/influxdb/v2/pkg/limiter"
+	"github.com/stretchr/testify/require"
 )
 
 func TestFixed_Available(t *testing.T) {
 	f := limiter.NewFixed(10)
-	if exp, got := 10, f.Available(); exp != got {
-		t.Fatalf("available mismatch: exp %v, got %v", exp, got)
-	}
+	require.Equal(t, 10, f.Available())
 
-	f.Take()
-
-	if exp, got := 9, f.Available(); exp != got {
-		t.Fatalf("available mismatch: exp %v, got %v", exp, got)
-	}
+	require.NoError(t, f.Take(context.Background()))
+	require.Equal(t, 9, f.Available())
 
 	f.Release()
+	require.Equal(t, 10, f.Available())
+}
+
+func TestFixed_Timeout(t *testing.T) {
+	f := limiter.NewFixed(1)
+	require.NoError(t, f.Take(context.Background()))
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
Does this mean that this test will always take 1 second to complete? If so maybe you could use a smaller unit of time to make the test go faster.